torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -55,14 +56,31 @@ std::unordered_set<Symbol> tuple_ops = {
 };
 
 struct ConstantPropagator {
-  ConstantPropagator(std::shared_ptr<Graph> graph)
-      : graph_(std::move(graph)), aliasDb(graph_){};
+  // Runs constant propagation with an aliasing db and checks if inputs
+  // might be mutated in the graph
+  static ConstantPropagator WithAliasDb(std::shared_ptr<Graph> graph) {
+    return ConstantPropagator(graph, true);
+  }
+
+  // Runs constant propagation only on ops with non-aliasing inputs
```

Comment:
this is a bit confusing as a comment, maybe: only on ops that clearly do not have aliased inputs even without computing aliasing information.